pkg/rkecerts/expiration.go

```diff
@@ -0,0 +1,59 @@
+package rkecerts
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+	"time"
+
+	rkeCluster "github.com/rancher/rke/cluster"
+	"github.com/rancher/rke/pki"
+	"github.com/rancher/rke/pki/cert"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	v1 "k8s.io/api/core/v1"
+)
+
+func CleanCertificateBundle(certs map[string]pki.CertificatePKI) {
+	for name := range certs {
+		if strings.Contains(name, "client") ||
+			strings.Contains(name, "token") ||
+			strings.Contains(name, "header") ||
+			strings.Contains(name, "admin") {
+			delete(certs, name)
+		}
+	}
+}
+
+func GetCertExpiration(c string) (v3.CertExpiration, error) {
+	date, err := GetCertExpirationDate(c)
+	if err != nil {
+		return v3.CertExpiration{}, err
+	}
+	return v3.CertExpiration{
+		ExpirationDate: date.Format(time.RFC3339),
+	}, nil
+}
+
+func GetCertExpirationDate(c string) (*time.Time, error) {
+	certs, err := cert.ParseCertsPEM([]byte(c))
+	if err != nil {
+		return nil, err
+	}
+	if len(certs) == 0 {
+		return nil, errors.New("certificate contains invalid dates")
+	}
+	return &certs[0].NotAfter, nil
+}
+
+func CertBundleFromConfig(cm *v1.ConfigMap) (map[string]pki.CertificatePKI, error) {
+	rawCerts, ok := cm.Data[rkeCluster.FullStateConfigMapName]
+	if !ok {
+		return nil, errors.New("full-cluster-state configmap does not contain data")
+	}
+	rkeFullState := &rkeCluster.FullState{}
+	err := json.Unmarshal([]byte(rawCerts), rkeFullState)
+	if err != nil {
+		return nil, err
+	}
+	return rkeFullState.DesiredState.CertificatesBundle, nil
```

Comment:
we should return Current state not the desired state since it reflects the current certificates actually deployed on the nodes